neuraxle/pipeline.py

```diff
@@ -64,26 +65,7 @@ def inverse_transform(self, processed_outputs):
         return self.inverse_transform_processed_outputs(processed_outputs)
 
 
-class ResumableStep(BaseStep):
-    """
-    A step that can be resumed, for example a checkpoint on disk.
-    """
-
-    """
-    Load Pipeline Checkpoint
-    :param data_inputs: initial data inputs
-    :return: steps_left_to_do, data_inputs
-    """
-    @abstractmethod
-    def load_checkpoint(self, data_inputs) -> Tuple[list, Any]:
-        raise NotImplementedError()
-
-    @abstractmethod
-    def should_resume(self, data_inputs) -> bool:
-        raise NotImplementedError()
-
-
-class Pipeline(BasePipeline, ResumableStep):
+class Pipeline(BasePipeline, BaseStep, ResumableStepMixin):
```

Comment:
No need to inherit from BaseStep because BasePipeline already inherints from it and it would cause a diamond problem.

Do: 
`class Pipeline(ResumableStepMixin, BasePipeline):`

Reply:
lol right thanks